devtools/aesh/src/main/java/io/quarkus/cli/commands/DevModeCommand.java

```diff
@@ -0,0 +1,160 @@
+    private File projectPath;
+
+    private BuildTool buildTool;
+    private File buildFile;
+
+    @Override
+    public CommandResult execute(CommandInvocation invocation) throws CommandException, InterruptedException {
+
+        if (!verifyProjectStatus(invocation))
+            return CommandResult.FAILURE;
+
+        try {
+            devMode(invocation);
+        } catch (InterruptedException i) {
+            invocation.println("Stopping dev-mode");
+        }
+
+        return CommandResult.SUCCESS;
+    }
+
+    private void devMode(CommandInvocation invocation) throws InterruptedException {
+        if (buildTool.equals(BuildTool.MAVEN)) {
+            File wrapper = ExecuteUtil.getMavenWrapper(projectPath.getAbsolutePath());
+            if (wrapper != null) {
+                ExecuteUtil.executeWrapper(invocation, wrapper, "quarkus:dev");
+            } else {
+                ExecuteUtil.executeMaven(projectPath, invocation, "quarkus:dev");
+            }
+
+        }
+        //do gradle
+        else {
+            File wrapper = ExecuteUtil.getGradleWrapper(projectPath.getAbsolutePath());
+            if (wrapper != null) {
+                ExecuteUtil.executeWrapper(invocation, wrapper, "quarkusDev");
+            } else {
+                ExecuteUtil.executeGradle(projectPath, invocation, "quarkusDev");
+            }
+        }
+    }
+
+    private boolean verifyProjectStatus(CommandInvocation invocation) {
+
+        if (projectPath == null)
+            projectPath = new File(System.getProperty("user.dir"));
+
+        buildTool = QuarkusProject.resolveExistingProjectBuildTool(projectPath.toPath());
+
+        if (buildTool.getBuildFiles().length > 0)
+            buildFile = new File(buildTool.getBuildFiles()[0]);
+
+        if (buildFile == null || !buildFile.isFile()) {
+            invocation.println("Was not able to find a build file in: " + projectPath);
+            return false;
+        }
+
+        if (buildDir == null)
+            buildDir = new File(buildTool.getBuildDirectory());
+
+        if (sourceDir == null)
+            sourceDir = resolveSourceDir();
+
+        if (!sourceDir.isDirectory()) {
+            invocation.println("ERROR: The project's sources directory does not exists (" + sourceDir);
+            return false;
+        }
+
+        if (!buildDir.isDirectory() || !new File(buildDir, "classes").isDirectory()) {
+            invocation.println("Build directory (" + buildDir + " wasn't found. Compiling...");
+            //if we run interactive, the name is 'compile-project', otherwise it's 'quarkus compile-project'
+            try {
+                invocation.executeCommand("compile-project");
+            } catch (CommandNotFoundException e) {
+                try {
+                    invocation.executeCommand("quarkus compile-project");
+                } catch (CommandNotFoundException | CommandLineParserException | OptionValidatorException
```

Comment:
There are no shared base class for these exceptions ?

Reply:
No, but that's a good idea, I'll implement that in Æsh asap